server/command.go

```diff
@@ -221,6 +221,22 @@ func (p *Plugin) executeLinkCommand(args *model.CommandArgs, parameters []string
 		return p.cmdError(args.UserId, args.ChannelId, "Error occurred while saving the subscription")
 	}
 
+	if _, err = p.API.ShareChannel(&model.SharedChannel{
+		ChannelId: channelLink.MattermostChannelID,
+		TeamId:    channelLink.MattermostTeamID,
+		Home:      true,
+		ReadOnly:  false,
+		CreatorId: p.userID,
+		RemoteId:  p.remoteID,
+		ShareName: channelLink.MattermostChannelID,
+	}); err != nil {
+		if err2 := p.store.DeleteSubscription(channelsSubscription.ID); err2 != nil {
+			p.API.LogDebug("Unable to rollback the subscription creation", "channelID", channelLink.MattermostChannelID, "error", err2.Error())
+		}
```

Comment:
Given we're not relying on this shared channels feature yet, I'm wondering if we shouldn't be destructive here and deleting the subscription?

```suggestion
```

Reply:
This is not "deleting the subscription", this is "undoing the channel linking done before in case of error"

Reply:
If shared channels was a subsystem we were relying on, I agree this code makes sense (though I'd suggest we invert and start sharing the channel first). But for the purpose of this PR -- an experiment to measure the overhead of shared channels -- we're directly tying the functionality of that core to the enduser's experience, breaking channel linking if shared channels doesn't work for some reason. Should we narrow the scope of this PR to "just an experiment" with 0 expected impact on the enduser?

cc @jespino / @wiggin77 as I sense this is a broader question for what we want to achieve here.